### PR TITLE
Use `F.matmul` instead of `F.batch_matmul` in memnn example

### PR DESCRIPTION
**Pull request by takagi**
This PR fixes the memnn example to use `F.matmul` rather than `F.batch_matmul`, which is deprecated now. With `F.matmul`, the memnn example can work in FP16 mode.

### CODE REVIEW

File: examples/memnn/memnn.py [hunk 1]
```diff
@@ -116,8 +116,8 @@ def query(self, u):
         tc = self.TC(inds)
         tm = F.broadcast_to(tm, self.m.shape)
         tc = F.broadcast_to(tc, self.c.shape)
-        p = F.softmax(F.batch_matmul(self.m + tm, u))
-        o = F.batch_matmul(F.swapaxes(self.c + tc, 2, 1), p)
+        p = F.softmax(F.matmul(self.m + tm, u[..., None]))
```
Comment (toslunar) [inline examples/memnn/memnn.py]: `F.expand_dims` performs better on backprop.

Comment (takagi) [reply examples/memnn/memnn.py]: Fixed!

File: examples/memnn/memnn.py [hunk 2]
```diff
@@ -116,8 +116,8 @@ def query(self, u):
         tc = self.TC(inds)
         tm = F.broadcast_to(tm, self.m.shape)
         tc = F.broadcast_to(tc, self.c.shape)
-        p = F.softmax(F.batch_matmul(self.m + tm, u))
-        o = F.batch_matmul(F.swapaxes(self.c + tc, 2, 1), p)
+        p = F.softmax(F.matmul(self.m + tm, F.expand_dims(u, u.ndim)))
```
Comment (toslunar) [inline examples/memnn/memnn.py]: How about this?
```suggestion
        p = F.softmax(F.matmul(self.m + tm, F.expand_dims(u, -1)))
```

Comment (takagi) [reply examples/memnn/memnn.py]: Sure!

**Comment by toslunar** [comment]
Jenkins, test this please.

**Pull request merged**